### PR TITLE
Fix Nil pointer references for WebSystemdSocket

### PR DESCRIPTION
**Pull request by ethan-gallant**
Noticed this when working with https://github.com/Lusitaniae/apache_exporter

If you don't provide a boolean value for WebSystemdSocket (default false) calling ListenAndServe causes the application to SigSegv. This checks if it is nil before evaluating the boolean value.

### CODE REVIEW

File: web/tls_config.go [hunk 1]
```diff
@@ -214,6 +214,11 @@ func ListenAndServe(server *http.Server, flags *FlagConfig, logger log.Logger) e
 		}
 		return ServeMultiple(listeners, server, flags, logger)
 	}
+
+	if flags.WebListenAddresses == nil {
+		*flags.WebListenAddresses = []string{":9090"}
+	}
```
Comment (SuperQ) [inline web/tls_config.go:220]: We definitely should not do this, as some users do not want any web listeners, only systemd socket.

**Review by SuperQ — CHANGES_REQUESTED** [review]
Maybe a better check would be to make sure that there is at least one valid listener and error if there is none.

File: web/tls_config.go (hunk 1 above)
Comment (dswarbrick) [reply web/tls_config.go:220]: I agree, in addition to it being highly presumptuous for a library (as opposed to an app) to choose a default listener address.

Comment (roidelapluie) [reply web/tls_config.go:220]: +1

Comment (ethan-gallant) [reply web/tls_config.go]: I definitely agree, I've implemented an error log instead. Cheers! :)

File: web/tls_config.go [hunk 2]
```diff
@@ -214,6 +214,11 @@ func ListenAndServe(server *http.Server, flags *FlagConfig, logger log.Logger) e
 		}
 		return ServeMultiple(listeners, server, flags, logger)
 	}
+
+	if flags.WebListenAddresses == nil {
+		return errors.New("no valid WebListenAddress specified")
```
Comment (roidelapluie) [inline web/tls_config.go:219]: Can we remove the if? WebListenAddress is not required

**Review by SuperQ — APPROVED** [review]
Perfect, thanks!

**Comment by ethan-gallant** [comment]
@SuperQ should be good to go. Squashed commits and amended to have the signature

File: web/tls_config.go [hunk 3]
```diff
@@ -203,7 +203,11 @@ func ServeMultiple(listeners []net.Listener, server *http.Server, flags *FlagCon
 // WebSystemdSocket in the FlagConfig is true. The FlagConfig is also passed on
 // to ServeMultiple.
 func ListenAndServe(server *http.Server, flags *FlagConfig, logger log.Logger) error {
-	if *flags.WebSystemdSocket {
+	if flags.WebSystemdSocket == nil && (flags.WebListenAddresses == nil || len(*flags.WebListenAddresses) == 0) {
+		return errors.New("no web listen address or systemd socket flag specified")
```
Comment (SuperQ) [inline web/tls_config.go:207]: Minor nit, I like to make these kinds of errors package vars so they can be used with `errors.Is()`.

Something like this:
```go
var ErrNoListeners = errors.New("...")
```

Comment (ethan-gallant) [reply web/tls_config.go]: I did a constant at the top if you're able to re-review :)

File: web/tls_config.go [hunk 4]
```diff
@@ -35,6 +35,8 @@ var (
 	errNoTLSConfig = errors.New("TLS config is not present")
 )
 
+const ErrNoListeners = errors.New("no web listen address or systemd socket flag specified")
```
Comment (SuperQ) [inline web/tls_config.go:38]: Huh, I didn't know that this was a valid expression. Is this a new Go feature?

Comment (SuperQ) [reply web/tls_config.go:38]: Looks like it's still not valid.

```
web/tls_config.go:38:24: errors.New("no web listen address or systemd socket flag specified") (value of type error) is not constant
```

Comment (ethan-gallant) [reply web/tls_config.go:38]: I just mashed it out in my IDE but didn't actually test it. I was running go 1.20 so it's possible it's a new thing.

I'll change it to a Var tomorrow.

Comment (ethan-gallant) [reply web/tls_config.go]: Should be good to go now @SuperQ

**Review by SuperQ — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**